### DISCUSSION TITLE
Where can I find poses?

### DISCUSSION THREAD

**Discussion by uhvardhan** [USC-ACTLab/crazyswarm#566] [Q&A]
Hi,

I have been using crazyswarm for the past week or so. I have been able to run most of the tutorials and example codes present.

I was wondering about the pose of each quadrotor. I have enabled pose_logging to True. Now I am wondering as to where I can find the relevant poses for each CF. 

I just wanted to calculate how much the CF deviates from the ground truth. 

P.S. I am using Loco Positioning System currently.

**Comment by jpreiss** [comment]
It gets broadcasted as a ros topic e.g. `cf1/pose` and through the ROS tf system: https://github.com/USC-ACTLab/crazyswarm/blob/26edf4b27e25d52f57f223fdc72da2c75080c061/ros_ws/src/crazyswarm/src/crazyswarm_server.cpp#L743-L767

This means it will be available through `cf.position()` in your script.

**This should be documented.**

@whoenig What happens when pose logging is enabled but a mocap system is also in use? This could occur when measuring the accuracy of a non-mocap positioning system. Per the linked code, won't the ROS tf jitter back and forth between the EKF estimate and the mocap?

**Reply by uhvardhan** [reply]
Thank you @jpreiss.

**Reply by whoenig** [reply]
I think having mocap and pose logging enabled would cause the jitter you mentioned and should be fixed. This should have no effect on correct operation, since tf is only used for visualization purposes, not for actual computations. Tracked in https://github.com/USC-ACTLab/crazyswarm/issues/572.
